playbooks/roles/rpc_pre_upgrade/tasks/main.yml

```diff
@@ -16,6 +16,36 @@
 # set backup_dir as a fact to ensure dir name doesn't change during run
 - set_fact:
     backup_dir: "{{ ansible_env.BKUPDIR |default(local_home+'/'+'rpc-upgrade-'+date_stamp) }}"
+    marker_dir: "{{ ansible_env.UPGRADE_LEAP_MARKER_FOLDER | default('/etc/openstack_deploy/upgrade-leap') }}"
+
+- file:
```

Comment:
Making dir is same as above. Only main task is executed or the directory not existed.